### PR TITLE
Main revamp

### PR DESCRIPTION
**Pull request by sabhari-krr**
- Detached dependency from WPLoyalty Core
- Changed the way of rendering Template. (Ref: Discount Rules v3)
- Changed the Input Helper (Ref: Discount Rules v3)

### CODE REVIEW

File: App/Conditions/Base.php [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+use DateTime;
+use Exception;
+use stdClass;
+use Wljm\App\Helpers\Input;
+use Wljm\App\Helpers\Woocommerce;
+abstract class Base {
+	public static $woocommerce_helper, $filter;
+	public $name = null, $rule = null, $label = null, $group = null, $input, $extra_params = array( 'render_saved_condition' => false );
+
+	function __construct() {
```
Comment (alagesanbe08) [inline App/Conditions/Base.php:20]: Don't do static variable object in constructor. WPLoyalty we made mistake, we have to fix in wployalty also in later.

$woocommerce_helper and $this->input. Need to change that

File: App/Conditions/Base.php [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+
+	/**
+	 * return the name of the condition. If condition does not have name, then the condition will not get consider.
+	 * @return null
+	 */
+	function name() {
+		return $this->name;
+	}
+
+	function getCalculateBased( $data = array() ) {
```
Comment (alagesanbe08) [inline App/Conditions/Base.php:35]: change array() -> []

File: App/Conditions/Base.php (hunk 1 above)
Comment (sabhari-krr) [reply App/Conditions/Base.php]: - Removed $woocommerce_helper and used the getInstance() of Woocommerce in all the required places.
- Removed the $this->input, and not added anywhere, as it nowhere used.

File: App/Conditions/Base.php (hunk 2 above)
Comment (sabhari-krr) [reply App/Conditions/Base.php]: Changed all the array() to []

**Comment by alagesanbe08** [comment]
No need

**Pull request closed without merging**